### PR TITLE
Fixed/improved e2e tests

### PR DESCRIPTION
**Pull request by chaosmail**
Ok, pretty much what we discussed today. Let me know what you think!

### CODE REVIEW

**Comment by chaosmail** [comment]
I reorganized the temporary directories, now everything gets built into `.tmp`, i hope thats ok. I want to create a `dist` or `build` directory and compile the library automatically when releasing a new version.

**Comment by chaosmail** [comment]
Another important change, I am seeing this very often. I introduced this convention:
* unit tests are now called as `spec` - it's an executable specification/documentation of our module 
* integration tests are now called e2e - we are testing the browser integration of the components

File: gulpfile.js [hunk 1]
```diff
@@ -3,111 +3,209 @@ var gulp   = require('gulp');
...
-  gulp.src('test/coverage/**/lcov.info')
+// Submit code coverage to Coveralls
+gulp.task('coveralls', ['clean:coverage'], function() {
+  gulp.src(paths.coverage.to + '**/lcov.info')
     .pipe($.coveralls());
 });
 
-gulp.task('default', ['build']);
+// Default task running `gulp`
+gulp.task('default', function(){
```
Comment (lorem--ipsum) [inline gulpfile.js]: Any specific reason you use a function here ?

File: test/e2e/series/Area.e2e.ts [hunk 2]
```diff
@@ -11,7 +13,7 @@ describe('Area series', function() {
     var lines = element.all(by.css('.chart .area-series'));
     expect(lines.count()).toBe(2);
 
-    expect(lines.get(0)).toHaveExactClass('area-series mySeries0');
-    expect(lines.get(1)).toHaveExactClass('area-series mySeries1');
+    expect(lines.get(0).getAttribute('class')).toBe('area-series mySeries0');
```
Comment (lorem--ipsum) [inline test/e2e/series/Area.e2e.ts]: Oh. Now I feel stupid. :-D

File: tsd.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "version": "v4",
-  "repo": "borisyankov/DefinitelyTyped",
+  "repo": "chaosmail/DefinitelyTyped",
```
Comment (lorem--ipsum) [inline tsd.json]: Did you fork the repo ?

File: gulpfile.js (hunk 1 above)
Comment (chaosmail) [inline gulpfile.js]: I had some problems, that the gulp task finished nut didnt exit properly. someone proposed this as a workaround but it didnt change anything. So i better change it back :)

File: test/e2e/series/Area.e2e.ts (hunk 2 above)
Comment (chaosmail) [inline test/e2e/series/Area.e2e.ts]: don't worry, i tried to use mocha and chai and had to do a similar approach like yours. however if you use jasmine in protractor the `toBe` methid can deal with promises out of the box ;)

File: tsd.json (hunk 3 above)
Comment (chaosmail) [inline tsd.json]: basically yes, the problem is that both angular-protractor and jquery use the global $ variable. this leads to a typescript conflict/error. I propsed a PR to DefinitelyTyped to allow a noConflict version of jquery - however it's not straight forward. therefore i forked the repo, using a noConflict version of jQuery

**Comment by chaosmail** [comment]
this webdriver error on travis is nasty

**Comment by lorem--ipsum** [comment]
Yup, I took a look at it... probably some weird config issue...

**Comment by chaosmail** [comment]
Ok seems like chromium is stuck in version 37 on travis [https://github.com/travis-ci/travis-ci/issues/3475](https://github.com/travis-ci/travis-ci/issues/3475). Now I installed the latest chrome version and it works fine!

**Comment by chaosmail** [comment]
Alright, I rebased. Let me know if you agree on all the changes.

**Comment by chaosmail** [comment]
Ok one more thing, the `watch` and `build` tasks are not running the integration tests - it is only included in the `travis` task. However, it can be executed via `gulp test:e2e` separately. The reason for this, is that I want to have fast linting and automative unit test (maybe we could even use to phantomJS) while developing + an extensive integration test suite.

**Comment by chaosmail** [comment]
I also added linting for the test files

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Pull request merged**